plugins/trace_api_plugin/include/eosio/trace_api/trace.hpp

```diff
@@ -21,11 +21,21 @@ namespace eosio { namespace trace_api {
       chain::bytes                        data = {};
    };
 
+      struct action_trace_v1 {
+      uint64_t                            global_sequence = {};
+      chain::name                         receiver = {};
+      chain::name                         account = {};
+      chain::name                         action = {};
+      std::vector<authorization_trace_v0> authorization = {};
+      chain::bytes                        data = {};
+      chain::bytes                        return_value = {};   // add return_value 
+   };
+
   struct transaction_trace_v0 {
       using status_type = chain::transaction_receipt_header::status_enum;
 
       chain::transaction_id_type    id = {};
-      std::vector<action_trace_v0>  actions = {};
+      std::vector<action_trace_v1>  actions = {};
    };
```

Comment:
Did you test reading an old `trace_api_plugin` file created before this change? I don't think it will work as the `transaction_trace_v0` format is changed here. It would be good to add a test where we read old supported versions of `trace_api_plugin`.

Reply:
As this plugin just commit for few months, there shouldn't be many old trace. so, This is a quick way to support new structure member return_value.  If we want have function of reading old trace and new trace, We have to define trace_block_v3. and then if any member added in future, we have to define trace_block_v4, and v5, v6 ..., and the process function would be more and more, do you think that is a better way to support return_value?      

Usually a version field is used to read different version, so as to deal different version in one function, but there is no version filed in action_trace and transaction_trace and also block_track, if you want support different version I can add a version field and then I think too much code will be changed.

Reply:
The plugin has already been released, so I would think we would want to support old file formats. That is a question for @b1bart

Reply:
That's OK, I am going to define a new block structure name block_tarce_v2, this way we touch the old code. I will give a new pull later.

Reply:
type wrong, I mean this way we don't touch the old code. just add new things.